subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ResolveContext.java

```diff
@@ -34,4 +37,12 @@ public interface ResolveContext {
 
     AttributeContainer getAttributes();
 
+    /**
+     * Returns the synthetic dependencies for this context. Synthetic dependencies are dependencies which
+     * are an internal implementation detail of Gradle, used for example in dependency locking or
+     * consistent resolution. They are not "real" dependencies in the sense that they are not added by
+     * users, and they are not always used during resolution based on which phase of execution we are
+     * (task dependencies, execution, ...)
+     */
+    List<? extends DependencyMetadata> getSyntheticDependencies();
```

Comment:
🤔 I know it's a preexisting name, but I think you should try and think up something more descriptive than "synthetic" dependencies for this.  I don't have any suggestions myself, I just feel like this name obscures.  Maybe name it more by purpose?

Reply:
Changing this to `getLockDependencyConstraints`.

Reply:
`Synthetic` was used when we introduced consistent resolution as those were no longer only locking related. Going back to locking ties it too much to a single concept IMO while there is actually more than one already.

Reply:
So both of types of synthetic dependencies, dependency locking and consistent resolution, pass in `true` for the `fromLock` of `LocalComponentDependencyMetadata`. This property gets used in `DependencyState#isFromLock`, and then finally in `SelectorState#update`. See: https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/SelectorState.java#L382-L385

From there it gets used in a few other places in the resolution engine which I am not familiar with. My reasoning for renaming away from synthetic is that it seems the `Lock` terminology is being used elsewhere to reference both of these types of synthetic dependency constraints. 

Now my question: Should the code in the resolution engine be updated to refer to these types of variables as `synthetic` OR is the resolution engine incorrectly also treating  consistent resolution constraints as lock dependencies when it shouldnt?

Reply:
From looking at code, I would say that the `fromLock` should be replaced and use something based on `synthetic`, as it indeed merges the locking and consistent resolution constraints.

Reply:
I can go and rename those methods. I was just worried that code using `fromLock` only intended to check if it was a locked dependency and not necessarily consistent resolution as well.

Reply:
On second thought, I am going to leave these as is. I am not comfortable enough around here to verify all this code truly cares if the dependencies are locked or synthetic.